torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -153,6 +158,7 @@ def _lazy_init(self) -> None:
             raise RuntimeError(
                 f"FSDP requires a single root module but got {self._modules}"
             )
+        detect_compiled_autograd()
```

Comment:
is this executing after we wrap the fsdp2 with torch.compile?

if so, we could use `torch._dynamo.config.compiled_autograd`

Reply:
https://github.com/pytorch/pytorch/pull/138241

Reply:
discussed in https://github.com/pytorch/pytorch/pull/138241 - we couldn't set `torch._dynamo.config.compiled_autograd=True` on automatically within the CA ctx manager, and thus won't be able to use it here